src/dfx/src/commands/canister/tail.rs

```diff
@@ -0,0 +1,88 @@
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::lib::operations::canister;
+use crate::lib::root_key::fetch_root_key_if_needed;
+use candid::Principal;
+use clap::Parser;
+use dfx_core::identity::CallSender;
+use ic_utils::interfaces::management_canister::FetchCanisterLogsResponse;
+use slog::info;
+
+/// Get the canister logs.
+#[derive(Parser)]
+pub struct TailOpts {
+    /// Specifies the name or id of the canister to get its canister information.
+    canister: String,
+}
+
+fn format_canister_logs(logs: FetchCanisterLogsResponse) -> Vec<String> {
+    logs.canister_log_records
+        .into_iter()
+        .map(|r| {
+            let time = chrono::NaiveDateTime::from_timestamp_nanos(r.timestamp_nanos as i64)
+                .expect("invalid timestamp");
+
+            let message = if let Ok(s) = String::from_utf8(r.content.clone()) {
+                if format!("{s:?}").contains("\\u{") {
+                    format!("{:?}", r.content)
+                } else {
+                    s
+                }
+            } else {
+                format!("{:?}", r.content)
+            };
+
+            format!(
+                "[{}. {}]: {}",
+                r.idx,
+                time.format("%Y-%m-%d %H:%M:%S%.3f"),
+                message
+            )
+        })
+        .collect()
+}
+
+#[test]
+fn test_format_canister_logs() {
+    use ic_utils::interfaces::management_canister::CanisterLogRecord;
+
+    let logs = FetchCanisterLogsResponse {
+        canister_log_records: vec![
+            CanisterLogRecord {
+                idx: 42,
+                timestamp_nanos: 1620328630000000000,
+                content: b"Some text message".to_vec(),
+            },
+            CanisterLogRecord {
+                idx: 43,
+                timestamp_nanos: 1620328630001000000,
+                content: vec![1, 2, 3],
+            },
+        ],
+    };
+    assert_eq!(
+        format_canister_logs(logs),
+        vec![
+            "[42. 2021-05-06 19:17:10.000]: Some text message".to_string(),
+            "[43. 2021-05-06 19:17:10.001]: [1, 2, 3]".to_string(),
```

Comment:
I agree with @adamspofford-dfinity's comment thow showing these in a hex string would be easier to work with than decimals. Perhaps also adding a prefix to indicate that these are raw bytes would help as well.

```suggestion
            "[43. 2021-05-06 19:17:10.001] (bytes): 0x010203".to_string(),
```

Reply:
Done.